plugins/shortcuts/src/api/LocalStoredShortcuts.ts

```diff
@@ -16,38 +16,52 @@
 
 import { pageTheme } from '@backstage/theme';
 import { v4 as uuid } from 'uuid';
-import { ShortcutApi } from './ShortcutApi';
-import { Shortcut } from '../types';
-import { StorageApi } from '@backstage/core-plugin-api';
-import Observable from 'zen-observable';
+import { ShortcutApi, type Shortcut } from '@backstage/plugin-shortcuts';
+import type { StorageApi } from '@backstage/core-plugin-api';
+import type { Observable } from '@backstage/types';
+import ObservableImpl from 'zen-observable';
 
 /**
  * Implementation of the ShortcutApi that uses the StorageApi to store shortcuts.
  *
  * @public
  */
 export class LocalStoredShortcuts implements ShortcutApi {
-  private readonly shortcuts: Observable<Shortcut[]>;
+  private shortcuts: Shortcut[];
+  private readonly subscribers = new Set<
+    ZenObservable.SubscriptionObserver<Shortcut[]>
+  >();
+
+  private readonly observable = new ObservableImpl<Shortcut[]>(subscriber => {
+    // forward the the latest value
+    subscriber.next(this.shortcuts);
+
+    this.subscribers.add(subscriber);
+    return () => {
+      this.subscribers.delete(subscriber);
+    };
+  });
 
   constructor(private readonly storageApi: StorageApi) {
-    this.shortcuts = Observable.from(
-      this.storageApi.observe$<Shortcut[]>('items'),
-    ).map(snapshot => snapshot.value ?? []);
+    this.shortcuts = this.storageApi.snapshot<Shortcut[]>('items').value ?? [];
+    this.storageApi.observe$<Shortcut[]>('items').subscribe({
+      next: next => {
+        this.shortcuts = next.value ?? [];
+        this.notifyChanges();
+      },
+    });
   }
 
-  shortcut$() {
-    return this.shortcuts;
+  shortcut$(): Observable<Shortcut[]> {
+    return this.observable;
   }
 
   get() {
-    return Array.from(
-      this.storageApi.snapshot<Shortcut[]>('items').value ?? [],
```

Comment:
this was the source of the but in https://github.com/backstage/backstage/issues/16817

The `UserSettingsStorage` `storageApi` would always return a [static value of presence 'unknown'](https://github.com/backstage/backstage/blob/80698c1de81dc0774b87c466cb8951c11d4acea4/plugins/user-settings/src/apis/StorageApi/UserSettingsStorage.ts#L166-L168)